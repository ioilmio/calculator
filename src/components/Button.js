import React from 'react';
import PropTypes from 'prop-types';
import '../styles/css/button.css';

const Button = ({
  value, color, wide, handleClick,
}) => {
  const wideClass = wide ? `${color} wide` : `${color}`;
  return (
    <button type="button" className={wideClass} onClick={() => handleClick(value)}>{value}</button>
  );
};
Button.propTypes = {
  value: PropTypes.string.isRequired,
  color: PropTypes.string,
  wide: PropTypes.bool.isRequired,
  handleClick: PropTypes.func.isRequired,
};

Button.defaultProps = { color: 'orange' };

export default Button;
