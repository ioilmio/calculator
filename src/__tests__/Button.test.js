import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Button from '../components/Button';

const handleClick = () => ({ total: '0', next: '0', operation: '' });

describe('Button', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Button value="0" handleClick={handleClick} wide />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('creates a Button component', () => {
    render(<Button value="0" handleClick={handleClick} wide />);
    const element = screen.getByText('0');
    expect(element).toBeInTheDocument();
  });
});
