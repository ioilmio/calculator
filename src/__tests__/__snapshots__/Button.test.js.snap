// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="orange wide"
  onClick={[Function]}
  type="button"
>
  0
</button>
`;
